modules/n1ql/pages/n1ql-language-reference/typefun.adoc

```diff
@@ -1,36 +1,225 @@
 = Type Functions
-:description: Type functions perform operations that check or convert expressions.
+:description: The following checks if the supplied expression is a (data type).
 :page-topic-type: concept
 
 [abstract]
 {description}
 
+NOTE: If any arguments to any of the following functions are [.out]`MISSING` then the result is also [.out]`MISSING` (i.e.
+no result is returned).
+Similarly, if any of the arguments passed to the functions are `NULL` or are of the wrong type (e.g.
+an integer instead of a string), then `NULL` is returned as the result.
+
 == Type-Checking Functions
 
-ISARRAY(expression)
+== ISARRAY(expression)
+
+=== Description
 
 Returns True if expression is an array, otherwise returns MISSING, NULL or false.
 
-ISATOM(expression)
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISARRAY(true) AS `boolean`,
+       ISARRAY(MISSING) AS `missing`,
+       ISARRAY(NULL) AS `null`,
+       ISARRAY(123) AS `number`,
+       ISARRAY("hello world") AS `string`,
+       ISARRAY([1, 2, 3]) AS `array`,
+       ISARRAY({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISATOM(expression)
+
+=== Description
 
 Returns True if expression is a Boolean, number, or string, otherwise returns MISSING, NULL or false.
 
-ISBOOLEAN(expression)
+=== Examples
+====
+[source,n1ql]
+----
+SELECT ISATOM(true) AS `boolean`,
+       ISATOM(MISSING) AS `missing`,
+       ISATOM(NULL) AS `null`,
+       ISATOM(123) AS `number`,
+       ISATOM("hello world") AS `string`,
+       ISATOM([1, 2, 3]) AS `array`,
+       ISATOM({"hello": "world"}) AS `object`;
+----
+
+[source,json]
+----
+[
+  {
+    "array": true,
+    "boolean": false,
+    "null": null,
+    "number": false,
+    "object": false,
+    "string": false
+  }
+]
+----
+====
+
+== ISBOOLEAN(expression)
+
+=== Description
```

Comment:
Needs separate Description, Arguments, and Return Value sections, as per functions template

```suggestion
=== Description

Checks if the supplied expression is a Boolean.

=== Arguments

expression:: [Required] The expression to check.

=== Return Value
```